lighthouse-core/lib/i18n/i18n.js

```diff
@@ -127,9 +127,7 @@ function lookupLocale(locale) {
  * @param {string} icuMessage
  * @param {Record<string, *>} [values]
  */
-function _preprocessMessageValues(icuMessage, values) {
-  if (!values) return;
```

Comment:
feels like this change should be accompanied by a test too if we want to change it here :)

Reply:
maybe we should change the typedef below (https://github.com/GoogleChrome/lighthouse/pull/9384/files#diff-4f100f14d281ddfaff2dc4466fc78054R167) to `Record<string, any>`?

I guess I'm just not sure why

> so that when sending no replacement object the function will still fire.

is desirable.

Like why not just `if (!values) return {}` if we want to convert away from `undefined`

Reply:
> I guess I'm just not sure why
> 
> > so that when sending no replacement object the function will still fire.
> 
> is desirable.
> 
> Like why not just `if (!values) return {}` if we want to convert away from `undefined`

I may be misinterpreting your question, but this came up when adding the test. `_preprocessMessageValues` does the "all ICU replacements in the string have a value provided" check, which works fine if there are multiple replacements to be done and one value (or more) is missing. However if the values object is forgotten completely, this line skips the check, even if the string has replacements that will need values.

It still ends up throwing, just in `intl-messageformat`, so if we are going to do a check here it made sense to do it for all strings that need replacements (instead of all strings that have `values` passed in with them).

Reply:
Oh, oh, oh, I misread the `parsed.elements.forEach` line to be `Object.values(values).forEach`, my bad!

yes, yes this makes perfect sense!

Still would love a test for forgetting the `values` completely, but I gotcha now! 👍 :)

Reply:
Done! I added a new test, and made the error message a bit more helpful.